python/docs/source/conf.py

```diff
@@ -177,7 +177,24 @@
 # Add any extra paths that contain custom files (such as robots.txt or
 # .htaccess) here, relative to this directory. These files are copied
 # directly to the root of the documentation.
-#html_extra_path = []
+redirects_dir = 'redirects'
+html_extra_path = [redirects_dir]
+
+os.makedirs(redirects_dir, exist_ok=True)
+for moved_page in ['', '.ml', '.mllib', '.sql', '.streaming']:
+    moved_file = f'pyspark{moved_page}.html'
```

Comment:
I think simply mapping it wouldn't work. For example,

new URL:

```
reference/api/pyspark.sql.functions.from_json.html?highlight=from_json#pyspark.sql.functions.from_json
```

old URL:

```
pyspark.sql.html?highlight=from_json#pyspark.sql.functions.from_json
```

Can we leverage other extensions such as https://gitlab.com/documatt/sphinx-reredirects & https://pypi.org/project/sphinx-reredirects/, and just go back to the root page by wildcards?

Reply:
Oh, I didn't know we'd want to redirect anchor links as well.

In that case we'll have to dive into JavaScript, because the `<meta>` redirect cannot touch anchor values and convert them into paths (and the linked extension cannot do so either, as far as I can tell).

I'll submit a patch shortly.